src/cljs/harja/tiedot/hallinta/tyokalut/talvihoitoreitti_tyokalu.cljs

```diff
@@ -0,0 +1,251 @@
+    {:onnistui ->HaeKayttajanOikeuksiaOnnistui
+     :epaonnistui ->HaeKayttajanOikeuksiaEpaonnistui
+     :paasta-virhe-lapi? true}))
+
+(extend-protocol tuck/Event
+
+  Muokkaa
+  (process-event [{talvihoitoreitti :talvihoitoreitti} app]
+    (let [;; Tarkista hallintayksikko
+          talvihoitoreitti (if (and (not= (get-in app [:talvihoitoreitti :valittu-hallintayksikko]) (:valittu-hallintayksikko talvihoitoreitti)))
+                             (do
+                               (hae-hallintayksikon-urakat (:valittu-hallintayksikko talvihoitoreitti))
+                               ;; Haetaan urakkahaun yhteydessä aina myös tiedot, että onko käyttäjällä oikeudet lisätä urakkaan API:n kautta toteumia :hae-urakat-lisaoikeusvalintaan
+                               (hae-kayttajan-kaytto-oikeudet)
+                               (reset! valittu-urakka nil)
+                               (-> talvihoitoreitti
+                                 (assoc :valittu-urakka nil)))
+                             talvihoitoreitti)
+          ;; Aseta urakoitsija
+          talvihoitoreitti (if (and (:valittu-urakka talvihoitoreitti) (get-in talvihoitoreitti [:valittu-urakka :urakoitsija]))
+                             (assoc talvihoitoreitti :suorittaja-nimi (get-in talvihoitoreitti [:valittu-urakka :urakoitsija :nimi]))
+                             talvihoitoreitti)]
+      (-> app
+        (assoc :mahdolliset-urakat (if (and
+                                         @nav/hallintayksikon-urakkalista
+                                         (not= +mahdolliset-urakat+ @nav/hallintayksikon-urakkalista))
+                                     @nav/hallintayksikon-urakkalista
+                                     +mahdolliset-urakat+))
+        (assoc :talvihoitoreitti talvihoitoreitti))))
+
+  Laheta
+  (process-event [{talvihoitoreitti :talvihoitoreitti} app]
+    (let [tulos! (tuck/send-async! ->LahetysOnnistui)
+          virhe! (tuck/send-async! ->LahetysEpaonnistui)
+          urakkaid (:id (get-in app [:talvihoitoreitti :valittu-urakka]))]
+
+      (.setTimeout js/window
+        #(let [data (koostettu-data app)]
+           (if urakkaid
+             (go
+               (let [params {:body (.stringify js/JSON (clj->js data))
+                             :content-type :json
+                             :accept :json}
+                     ;; Lähetä talvihoitoreitti rajapintaan
+                     vastaus (<! (http/put (str "api/urakat/" urakkaid "/talvihoitoreitti") params))
+                     _ (println "Laheta :: vastaus" (pr-str vastaus))]
+                 (if (or (k/virhe? vastaus) (= 400 (:status vastaus)))
+                   (virhe!)
+                   (tulos!))))
+             (js/console.log "Urakkaa ei valittu! ei lähetetä (käyttöliittymä on vaikea, eli yritä uusiksi)")))
+
+        500)
+
+      (assoc app :lahetys-kaynnissa? true)))
+
+  LahetysOnnistui
+  (process-event [{vastaus :vastaus} app]
```

Comment:
-> ```_ app``` 

Nämä on vaan ehdotuksia, saa tuo vastaus destruktointi tuossa ollakin jos näissä halutaan kertoa mitä sieltä mahdollisesti  tulee takaisin tai jos tätä halutaan joskus käyttää/logittaa. Mutta yleensä näitä turhia parametreja ei tarvitse definata jos niillä ei mitään tehdä